src/pubkey.cpp

```diff
@@ -4,19 +4,182 @@
 
 #include "pubkey.h"
 
-#include "eccryptoverify.h"
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
 
-#include "ecwrapper.h"
+namespace
+{
+secp256k1_context* secp256k1_context_verify = NULL;
```

Comment:
Ok. But verify_refcount is not updated atomically, so multiple threads creating/destroying ECCVerifyHandles at the same time can cause problems. Maybe that's not supposed to happen though.